### PR TITLE
Include HasAltitude, HasBearing, HasSpeed, and HasAccuracy in Position

### PR DESCRIPTION
**Pull request by jixer**
Fixes # 310.

Changes Proposed in this pull request:
- Include HasAltitude, HasBearing, HasSpeed, and HasAccuracy in Position object
- Modify iOS's valid speed & bearing checks from `> -1` to `>= 0` _(per Apple docs mentioning invalid values being any negative number and the data type is double)_

&nbsp;

Implementation approach:
- HasLongitudeLatitude
  - Android
    - assign: true always
    - from Google docs: All locations generated by the LocationManager will have a valid latitude.
    - from Google docs: All locations generated by the LocationManager will have a valid longitude.
  - iOS
    - assign: CLLocation::horizontalAccuracy >= 0
    - from Apple docs: A negative value indicates that the latitude and longitude are invalid.
  - UWP
    - assign: true always
    - from MS docs: Values for the Latitude, Longitude, and Accuracy properties are always provided

- HasAltitude
  - Android
    - assign: (bool)Location::hasAltitude()
  - iOS
    - CLLocation::verticalAccuracy >= 0
    - from Apple docs: When this property contains a negative number, the value in the altitude property is invalid.
  - UWP
    - assign: Geoposition::Coordinate::Point::AltitudeReferenceSystem != AltitudeReferenceSystem.Unspecified
    - This one is a bit cryptic to me based upon the docs
    - from MS docs for Altitude: This field has an effect only if you set the altitude reference system of class that you create by using this BasicGeoPosition (For example: The AltitudeReferenceSystem property of a Geopoint ).

- HasAccuracy
  - Android
    - assign: (bool)Android::hasAccuracy()
  - iOS
    - assign: Always true
  - UWP
    - assign: true always
    - from MS docs: Values for the Latitude, Longitude, and Accuracy properties are always provided

- HasHeading
  - Android
    - assign: (bool)Location::hasBearing()
  - iOS
    - assign: (bool) CLLocation::course >= 0
    - from Apple docs: A negative value indicates that the course information is invalid.
  - UWP
    - assign: Geoposition::Coordinate::Heading != null
    - from MS docs: Values for the Altitude, AltitudeAccuracy, Heading, and Speed properties are provided if available

- HasSpeed
  - Android
    - assign: (bool)Location::hasSpeed()
  - iOS
    - assign: (bool) CLLocation::speed >= 0
    - from Apple docs: A negative value indicates an invalid speed
  - UWP
    - assign: Geoposition::Coordinate::Speed != null
    - from MS docs: Values for the Altitude, AltitudeAccuracy, Heading, and Speed properties are provided if available

### CODE REVIEW

**Comment by jixer** [comment]
One additional note is that the previous value assignments for Speed, Heading, etc are unchanged with this PR.  Therefore, there should be no breaking changes here and developers can migrate to use the new "Has" property booleans if and when they need them.

_EDIT_
The value assignments for speed & bearing are different for iOS with the change to utilize `>=0` rather than `> -1`.  It is assumed that this won't result in a change in functionality, but the change better aligns our code with Apple's documentation

File: src/Geolocator.Plugin/Apple/GeolocatorImplementation.apple.cs [hunk 1]
```diff
@@ -512,26 +517,36 @@ void OnLocationsUpdated(object sender, CLLocationsUpdatedEventArgs e)
 		void UpdatePosition(CLLocation location)
 		{
 			var p = (lastPosition == null) ? new Position() : new Position(this.lastPosition);
+			p.HasAccuracy = true;
 
-			if (location.HorizontalAccuracy > -1)
+			if (location.HorizontalAccuracy >= 0)
```
Comment (rohitvipin) [inline src/Geolocator.Plugin/Apple/GeolocatorImplementation.apple.cs]: Not sure why this needs to be changed?

Comment (jixer) [reply src/Geolocator.Plugin/Apple/GeolocatorImplementation.apple.cs]: I just put it in there as it seemed more appropriate based on Apple docs.  They mention an error as any negative number, which technically cloud also include numbers between 0 and -1.

Ultimately, it doesn't correspond to the goal of the pull request, so I'll yank it.

**Comment by jixer** [comment]
@jamesmontemagno This PR was closed without merge. Is there something I should do differently in my next PR?  Hoping to knock out issue #311.

**Comment by jamesmontemagno** [comment]
May i also recommend you PR to github.com/xamarin/essentials :)

**Comment by jamesmontemagno** [comment]
oh wait, why did that happen....

**Comment by jamesmontemagno** [comment]
So, i am wondering... should we just make these nullable? instead of "HasX"

**Comment by jixer** [comment]
@jamesmontemagno Sorry for the delayed response on this.  I didn't go the Nullable route in case we wanted to avoid a potentially breaking change (type mismatch) in people's code when they update (e.g., `double myLat = position.Latitude`).  That being said, I was probably just being unnecessarily cautious and conceptually nullable is much more intuitive.  I'll work on porting over.

**Comment by gmarbury** [comment]
@jamesmontemagno  and @jixer  as a suggestion you can use double.NaN if you are concerned with avoiding potential breaking changes with double? = null. We still have to check NaN too anyway. But what you have now looks very good. Please resolve the conflicts and get this checked in.

**Pull request closed without merging**